### ISSUE TITLE
Turn on CRDB changefeeds -> s3 for all prod environments

### ISSUE THREAD

**Issue by benesch** [MaterializeInc/materialize#27162]

**Comment by benesch** [comment]
From discussions with @danhhz and @pH14:

  * Paul has an old PR for this: https://github.com/MaterializeInc/cloud/pull/4555
  * One pain point in the past was setting up IAM delegation for this instead of using static access keys; plausibly this was fixed when we set up IAM delegation for backups and restores
  * Another blocker was that `SHOW CHANGEFEED JOBS` took like literal minutes to complete. I think this is fixed in v23.1 due to some optimizations they made to not scan the whole jobs table to compute `SHOW CHANGEFEED JOBS`.

tl;dr is that I'll try to find some time this quarter to revive Paul's PR and see if I can get it across the finish line.
